src/main/java/org/prebid/server/bidder/sharethrough/SharethroughBidder.java

```diff
@@ -1,244 +1,198 @@
-        } catch (IllegalArgumentException e) {
-            errors.add(BidderError.badInput(e.getMessage()));
-            return null;
-        }
+    private ExtSource updateExtSource(ExtSource ext) {
+        ext.addProperty("str", TextNode.valueOf(ADAPTER_VERSION));
+        ext.addProperty("version", TextNode.valueOf(prebidVersionProvider.getNameVersionRecord()));
+        return ext;
+    }
 
-        final SharethroughRequestBody body = strUriParameter.getBody();
+    private ExtSource createExtSource() {
+        final ExtSource extSource = ExtSource.of(null);
+        extSource.addProperty("str", TextNode.valueOf(ADAPTER_VERSION));
+        extSource.addProperty("version", TextNode.valueOf(prebidVersionProvider.getNameVersionRecord()));
+        return extSource;
+    }
 
-        return HttpRequest.<SharethroughRequestBody>builder()
+    private HttpRequest<BidRequest> makeHttpRequest(BidRequest request) {
+        return HttpRequest.<BidRequest>builder()
                 .method(HttpMethod.POST)
-                .uri(uri)
-                .headers(headers)
-                .body(mapper.encodeToBytes(body))
-                .payload(body)
+                .uri(endpointUrl)
+                .headers(HttpUtil.headers())
+                .payload(request)
+                .body(mapper.encodeToBytes(request))
                 .build();
     }
 
     @Override
-    public Result<List<BidderBid>> makeBids(HttpCall<SharethroughRequestBody> httpCall, BidRequest bidRequest) {
+    public final Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
         try {
-            final String responseBody = httpCall.getResponse().getBody();
-            final ExtImpSharethroughResponse sharethroughBid = mapper.mapper().readValue(responseBody,
-                    ExtImpSharethroughResponse.class);
-            return Result.withValues(toBidderBid(responseBody, sharethroughBid, httpCall.getRequest()));
-        } catch (IOException | IllegalArgumentException e) {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            final Video videoType = bidRequest.getImp().get(0).getVideo();
+            return Result.withValues(extractBids(bidResponse, videoType));
+        } catch (DecodeException | PreBidException e) {
             return Result.withError(BidderError.badServerResponse(e.getMessage()));
         }
     }
 
-    /**
-     * Converts {@link ExtImpSharethroughResponse} to {@link List} of {@link BidderBid}.
-     */
-    private List<BidderBid> toBidderBid(String responseBody, ExtImpSharethroughResponse sharethroughBid,
-                                        HttpRequest<SharethroughRequestBody> request) {
-        if (sharethroughBid.getCreatives().isEmpty()) {
-            throw new IllegalArgumentException("No creative provided");
+    private static List<BidderBid> extractBids(BidResponse bidResponse, Video videoType) {
```

Comment:
Accept `BidType` instead of `Video` object.

Reply:
We get `Video` from `Imp`, how we can get BidType?